### DISCUSSION TITLE
ページングやメモリ管理に関連する実験テーマがないか

### DISCUSSION THREAD

**Discussion by keisuke713** [nuta/microkernel-book#18] [General]
実行可能ファイルがメモリにロードされる際に以下のあたりで新しくメモリを確保して仮想アドレスと物理アドレスを紐付けて〜など処理をしているのは何となく分かるのですがコードを読んでいるだけでは腑に落ちないので、手を動かしながら学べたらなと思っています。
そこでページングやメモリ管理に関連する実験テーマがありましたら教えて頂きたいです。

https://github.com/nuta/microkernel-book/blob/main/servers/vm/page_fault.c#L60-L86
https://github.com/nuta/microkernel-book/blob/main/kernel/riscv32/vm.c#L62-L89

**Comment by keisuke713** [comment]
[こちら](https://github.com/nuta/microkernel-book/blob/main/IDEAS.md)に載っている `共有メモリの実装` は `既存のシステムコールを使う` と書かれているのでページングやメモリ管理とは少し方向性の違うテーマですよね？

**Reply by nuta** [reply]
[ページング関連のアイデア](https://github.com/nuta/microkernel-book/commit/c46c2de86f3c917eefc2b86f2a440b4b80c0fefa) を2つ追加してみました。

「共有メモリの実装」もメモリ管理関連のアイデアです！ただし、カーネルへの変更がなくとも実装できそうなので、カーネルに手を加えたいのであれば、新しいシステムコールとして共有メモリを実装するのも手です。「既存のシステムコールを使う」は手法の一つです :)

**Reply by keisuke713** [reply]
追加してくださりありがとうございます。
早速やってみようと思います。

> カーネルへの変更がなくとも実装できそうなので、カーネルに手を加えたいのであれば、新しいシステムコールとして共有メモリを実装するのも手です。

なるほど、システムコールとして実装する方法もあるのですね。面白そうなのでこちらも着手してみようと思います！
